### DISCUSSION TITLE
Wrong number of initial parameters

### DISCUSSION THREAD

**Discussion by anjankatna** [TheoMichelot/moveHMM#49] [Q&A]
Hi,
Is there a way of identifying the initial parameter values. I used the default given in the code. But can i use my own step length and angle data to determine the initial values. I would like to have three states - resting, foraging, and transit. My species are mesocarnivores (e.g. jungle cat).
I also received this error when preparing the model:

I used the vignette and prepared the following:

mu0 <- c(0.1,1) # step mean (two parameters: one for each state)
sigma0 <- c(0.1,1) # step SD
zeromass0 <- c(0.1,0.05,) # step zero-mass
stepPar0 <- c(mu0,sigma0,zeromass0)
angleMean0 <- c(pi,0) # angle mean
kappa0 <- c(1,1) # angle concentration
anglePar0 <- c(angleMean0,kappa0)

I then called the function
## call to fitting function
m1.hmm <- fitHMM(data = jc.jan18.HMM, nbStates = 2, stepPar0=stepPar0,
            anglePar0=anglePar0,formula=~landuse)

Then received the following error:
Error in fitHMM(data = jc.jan18.HMM, nbStates = 2, stepPar0 = stepPar0,  : 
  Wrong number of initial parameters -- there should be 4 initial step parameters -- there should be 4 initial angle parameters.

**Comment by anjankatna** [comment]
the term zeromass0 was creating an issue. i removed it and the model worked!

**Comment by anjankatna** [comment]
However, would really appreciate if I can get some information on selecting the initial parameters.

**Comment by dibeyp** [comment]
It looks like you aren't supplying the right number of parameters as per
the error message. I don't have R in front of me but you appear to be
passing in 6 step parameters when the function is asking for 4. I would
look at the zero mass parameter documentation carefully.

There is never a default way to identify starting values for maximum
likelihood estimation. A degree of trial and error is involved. I would get
rid of the error about parameter number first and see how the model
performs. Try a few different starting values. If the model converges to
the same estimates regardless, that is usually a good sign.

Hth
Toby

**Comment by anjankatna** [comment]
Thanks Toby,

Yes i got rid of the zero mass parameter and used values after looking at my step lengths and histogram. I did get my model results. Could you point me to a source that would help me infer these results. 

Additionally, when I use plot(m1.hmm), i get the error:
Error in if (max(stepDensities[[state]][, 2]) > maxdens) maxdens <- max(stepDensities[[state]][,  : 
  missing value where TRUE/FALSE needed.

Is there an additional option that should be used while plotting?

Also, is there an option using which I can segregate my actual data based on the two states - i.e. rows x,y,z = state 1, rows a,b,c = state 2.

Regards,
Anjan

**Comment by TheoMichelot** [comment]
Hi Anjan,

In addition to Toby's advice, there is a vignette specifically about selecting initial parameter values: [A short guide to choosing initial parameter values for the estimation in moveHMM](https://cran.r-project.org/web/packages/moveHMM/vignettes/moveHMM-starting-values.pdf). I recommend going through Sections 3 and 4 in particular.

The error that you get from `plot` probably means that the model didn't fit (i.e., the optimisation didn't converge), which could have resulted in infinite estimated parameters, for example.

I hope this helps, let me know if you still have any issues.

Théo

**Comment by anjankatna** [comment]
Thanks Theo and Toby,

I have indeed started reading the guide and other lit on the same. I run some pilot analysis and get back if there are issues,

Thanks very much for the quick responses and guidance.

Regards,
Anjan

**Comment by anjankatna** [comment]
Used the guide as suggested and did get my results! thank you
